Applications/Spire/Source/OrderImbalanceIndicator/OrderImbalanceIndicatorContextMenu.cpp

```diff
@@ -0,0 +1,226 @@
+        m_notional_value_signal(state == Qt::Checked);
+      });
+  m_table_columns_menu->addAction(notional_value_action);
+  auto export_table_action = new QAction(tr("Export Table"), this);
+  connect(export_table_action, &QAction::triggered,
+    [=] { m_export_table_signal(); });
+  addAction(export_table_action);
+  auto export_chart_action = new QAction(tr("Export Chart"), this);
+  connect(export_chart_action, &QAction::triggered,
+    [=] { m_export_chart_signal(); });
+  addAction(export_chart_action);
+  auto reset_action = new QAction(tr("Reset All Filters"), this);
+  connect(reset_action, &QAction::triggered, [=] { m_reset_signal(); });
+  addAction(reset_action);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_export_table_signal(
+    const SelectedSignal::slot_type& slot) const {
+  return m_export_table_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_export_chart_signal(
+    const SelectedSignal::slot_type& slot) const {
+  return m_export_chart_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_reset_signal(
+    const SelectedSignal::slot_type& slot) const {
+  return m_reset_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_security_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_security_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_side_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_side_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_size_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_size_signal.connect(slot);
+}
+
+connection
+    OrderImbalanceIndicatorContextMenu::connect_reference_price_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_ref_px_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_date_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_date_signal.connect(slot);
+}
+
+connection OrderImbalanceIndicatorContextMenu::connect_time_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_time_signal.connect(slot);
+}
+
+connection
+    OrderImbalanceIndicatorContextMenu::connect_notional_value_toggled_signal(
+    const ToggledSignal::slot_type& slot) const {
+  return m_notional_value_signal.connect(slot);
+}
+
+bool OrderImbalanceIndicatorContextMenu::eventFilter(QObject* watched,
+    QEvent* event) {
+  if(watched == m_table_columns_menu) {
+    if(event->type() == QEvent::Show) {
+      m_table_columns_menu->move(pos().x() + scale_width(136),
+        pos().y() + scale_height(6));
+    }
+  }
+  return QMenu::eventFilter(watched, event);
+}
+
+void OrderImbalanceIndicatorContextMenu::add_check_box(const QString& text,
+    QWidgetAction* action) {
+  auto check_box = new CheckBox(text, this);
```

Comment:
Use `std::unique_ptr`, then call check_box.release() where transferring ownership to `action`.
Also, `check_box` is only used at the end of the function, consider moving its initialization there.

Reply:
Seems like it also conflicts with Qt ownership model.

Reply:
Moving ```check_box``` to the end is a good idea, the scope of a variable should be minimized as much as possible. As for ownership, the ```CheckBox``` is owned by ```this```, which is fine.